src/librustc_typeck/astconv.rs

```diff
@@ -651,11 +651,16 @@ impl<'o, 'gcx: 'tcx, 'tcx> AstConv<'gcx, 'tcx>+'o {
         });
         let existential_projections = projection_bounds.iter().map(|bound| {
             bound.map_bound(|b| {
-                let p = b.projection_ty;
+                let trait_ref = self.trait_ref_to_existential(b.projection_ty.trait_ref(tcx));
+                let item_name = tcx.associated_item(b.projection_ty.item_def_id).name;
+                let item_def_id = tcx.associated_items(trait_ref.def_id).find(
+                    |item| item.name == item_name && item.kind == ty::AssociatedKind::Type
+                ).unwrap().def_id;
```

Comment:
I'm not entirely sure what `trait_ref_to_existential` does, see https://github.com/rust-lang/rust/blob/1d2db7b9e8082f0459e000985d77fc7ad0dabade/src/librustc/ty/subst.rs#L562